src/generator/AutoRest.CSharp.Azure.Fluent/CodeGeneratorCsaf.cs

```diff
@@ -0,0 +1,113 @@
+    // Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoRest.Core;
+using AutoRest.Core.Model;
+using AutoRest.CSharp.Azure.Model;
+
+using AutoRest.CSharp.Azure.Templates;
+using AutoRest.CSharp.Model;
+
+using AutoRest.CSharp.Templates;
+using AutoRest.Extensions.Azure;
+
+namespace AutoRest.CSharp.Azure.Fluent
+{
+    public class CodeGeneratorCsaf : CodeGeneratorCsa
+    {
+        private const string ClientRuntimePackage = "Microsoft.Rest.ClientRuntime.Azure.3.2.0";
+
+        /// <summary>
+        /// Generates C# code for service client.
+        /// </summary>
+        /// <param name="cm"></param>
+        /// <returns></returns>
+        public override async Task Generate(CodeModel cm)
+        {
+            var codeModel = cm as CodeModelCsa;
+            if (codeModel == null)
+            {
+                throw new InvalidCastException("CodeModel is not a Azure c# CodeModel");
+            }
+
+            // Service client
+            var serviceClientTemplate = new AzureServiceClientTemplate { Model = codeModel };
+            await Write(serviceClientTemplate, $"{codeModel.Name}{ImplementationFileExtension}");
+
+            // Service client interface
+            var serviceClientInterfaceTemplate = new ServiceClientInterfaceTemplate { Model = codeModel };
+            await Write(serviceClientInterfaceTemplate, "I" + codeModel.Name + ImplementationFileExtension);
+
+            // Operations
+            foreach (MethodGroupCs group in codeModel.Operations)
```

Comment:
It's actually not my code... I believe that it's still 'in development' .. and generally I didn't want to tweak too much. 

I think I'm going to have to wait and see if this doesn't change much before they are done with it. I get the feeling it's still a looooong way to go.